src/coordinated_workers/interfaces/cluster.py

```diff
@@ -306,6 +306,14 @@ def gather_addresses_by_role(self) -> Dict[str, Set[str]]:
                     continue
         return data
 
+    def gather_addresses_by_unit(self) -> Dict[str, Set[str]]:
```

Comment:
why do we care about the unit names? just to give names to the upstreams in the nginx configuration?